language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ForeachCompletionUtil.java

```diff
@@ -111,19 +144,29 @@ private static LSCompletionItem getForEachCompletionItem(BallerinaCompletionCont
                 .append(CommonUtil.getValidatedSymbolName(ctx, VAR_NAME)).append(" in ").append(symbolName)
                 .append(" ").append("{").append(CommonUtil.LINE_SEPARATOR).append("\t${1}")
                 .append(CommonUtil.LINE_SEPARATOR).append("}");
+        return ForeachCompletionItemBuilder.build(snippet.toString(), label, detail, textEdits);
+    }
 
-        //create the text edit item to replace the field accession text
-        List<TextEdit> textEdits = new ArrayList<>();
-        TextEdit textEdit = new TextEdit();
-        textEdit.setNewText("");
-        LineRange lineRange = expr.lineRange();
-        Position start = new Position(lineRange.startLine().line(), lineRange.startLine().offset());
-        Position end = new Position(lineRange.endLine().line(), lineRange.endLine().offset());
-        textEdit.setRange(new Range(start, end));
-        textEdits.add(textEdit);
-        CompletionItem completionItem =
-                ForeachCompletionItemBuilder.build(snippet.toString(), label, detail, textEdits);
-        return new StaticCompletionItem(ctx, completionItem, StaticCompletionItem.Kind.OTHER);
+    /**
+     * Returns the foreach completion item with range expression.
+     *
+     * @param ctx        completion context
+     * @param symbolName symbol name corresponding to the symbol
+     * @param symbol     type symbol
+     * @param textEdits  edits that replace the field access symbol
+     * @return
+     */
+    private static CompletionItem getRangeExprCompletionItem(BallerinaCompletionContext ctx,
+                                                             String symbolName, TypeSymbol symbol,
+                                                             List<TextEdit> textEdits) {
+        String detail = "foreach range expression snippet for iterable variable - " + symbolName;
+        String label = "foreach i - " + symbolName;
```

Comment:
Shall we avoid appending the symbol name here?

Reply:
Done!